src/tests/baseservices/exceptions/stackoverflow/stackoverflowtester.cs

```diff
@@ -4,27 +4,45 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 using System.Text;
 
 namespace TestStackOverflow
 {
     class Program
     {
+        const string ThisProjectName = "stackoverflowtester";
         static string s_corerunPath;
         static string s_currentPath;
 
-        static bool TestStackOverflow(string testName, string testArgs, out List<string> stderrLines)
+        static Process GetNativeAotProcess(string testName, string testArgs)
         {
-            Console.WriteLine($"Running {testName} test({testArgs})");
-            List<string> lines = new List<string>();
-
             Process testProcess = new Process();
+            testProcess.StartInfo.FileName = Environment.ProcessPath;
+            testProcess.StartInfo.Arguments = testArgs;
+            testProcess.StartInfo.Environment.Add("DOTNET_DbgEnableMiniDump", "0");
+            return testProcess;
+        }
 
-            testProcess.StartInfo.FileName = s_corerunPath;
-            testProcess.StartInfo.Arguments = $"{Path.Combine(s_currentPath, "..", testName, $"{testName}.dll")} {testArgs}";
+        static bool TestStackOverflow(string testName, string testArgs, out List<string> stderrLines, out bool checkStackFrame)
+        {
+            Console.WriteLine($"Running with args: {testName} {testArgs}");
+            testArgs = $"{testName} {testArgs}";
+            List<string> lines = new List<string>();
+
+            Process testProcess;
+            if (TestLibrary.Utilities.IsNativeAot)
+            {
+                testProcess = GetNativeAotProcess(testName, testArgs);
+            }
+            else {
+                testProcess  = new Process();
```

Comment:
A nit - it would be nice to create the process for both native aot and non-native aot inline or in separate function (or maybe just use a single function for the process creation) instead of doing it inline for one and in a function for the other case.

Reply:
Moved them both to be inline.